files/servicenow.rb

```diff
@@ -85,15 +91,34 @@ def response
 end
 
 def servicenow(certname)
-  config = YAML.load_file('/etc/puppetlabs/puppet/servicenow.yaml')
-
-  instance          = config['instance']
-  username          = config['user']
-  password          = config['password']
-  table             = config['table']
-  certname_field    = config['certname_field']
-  classes_field     = config['classes_field']
-  environment_field = config['environment_field']
+  hiera_eyaml_config = nil
+  begin
+    # Note: If hiera-eyaml config doesn't exist, then load_config_file returns
+    # the hash {:options => {}, :sources => []}
+    hiera_eyaml_config = Hiera::Backend::Eyaml::Subcommand.load_config_file
+  rescue StandardError => e
+    raise "error reading the hiera-eyaml config: #{e}"
+  end
+
+  servicenow_config = File.read('/etc/puppetlabs/puppet/servicenow.yaml')
+  unless hiera_eyaml_config[:sources].empty?
+    # hiera-eyaml config exists so decrypt the servicenow_config. Note that
+    # Decrypt.execute leaves plain-text values alone so this code should
+    # work OK even if the user specified plain-text config values.
+    Hiera::Backend::Eyaml::Options.set(hiera_eyaml_config[:options])
+    Hiera::Backend::Eyaml::Options[:source] = :file
+    Hiera::Backend::Eyaml::Options[:input_data] = servicenow_config
+    servicenow_config = Hiera::Backend::Eyaml::Subcommands::Decrypt.execute
```

Comment:
I thought about copy-pasting [Decrypt.execute's](https://github.com/voxpupuli/hiera-eyaml/blob/v3.2.0/lib/hiera/backend/eyaml/subcommands/decrypt.rb#L56) code in case its implementation changes, but that seems unlikely given that it was last updated 2 years ago, that the README talks about the `decrypt` command (making it public API), and that I also don't know how much of that execute's code uses private-API methods (e.g. is ParserFactory public or an implementation detail?).

If any of my assumptions turn out wrong in the future, then copy-pasting is easy enough so I opt to do it if it is needed.

Reply:
Ignore this, I went with a better way inspired by https://github.com/puppetlabs/puppet/blob/master/lib/puppet/functions/eyaml_lookup_key.rb#L80